### PR TITLE
Add Problem, Solution, and Impact Sections to Project Pages

### LINKED ISSUE #3547: Add Problem, Solution, and Impact Sections to Project Pages
### Overview
We need to design an additional section in the project page code that displays the Problem, Solution, and Impact & SDG info on each individual project page so that development can build the content into the site. 

### Details
<details><summary>Click here to see screenshot of proposed Figma design</summary>

image

</details>

View the type of  page that new content layout will go on https://www.hackforla.org/projects/100-automations

### Action Items
Note that you will only be updating one HTML file: _layouts/project.html). All project pages are generated from this file. There is not a separate file for every project
 - [x]  Examine the existing project page
 - [x]  Examine the Figma page for this issue and implement the code changes
 - [x] In between the `Project Description Card` and `Project Resources Card` sections in _layouts/project.html apply the following logic:
   - [x] Create a new div for `Project Value Card`
   - [x] Use Liquid to make an if-statement that will render the Project Value Card if the project file being called in /_projects has data listed under `problem`
     - [x] i.e. If `project.problem` exists, render the Project Value Card with all the information in `project.problem`, `project.solution` and `project.impact`
     - [x] Otherwise, do not render the `Project Value Card`
 - [ ] Submit a pull request.

### Resources/Instructions
- [Figma - Individual Project Page](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=7891%3A0)
- The original design issue: #3034 

#### Project page
-  see [project.html page](https://github.com/hackforla/website/blob/gh-pages/_layouts/project.html) on our repo
- location of project page in code base: _layouts/project.html
-  Example project page: https://www.hackforla.org/projects/100-automations (they are all made automatically from project.html and the project.md file)

### PR DESCRIPTION
**Pull request by patrickohh**
Fixes #3547

### What changes did you make and why did you make them ?

  - Added value card to project pages.
  - Value card only renders if specific project page has problem, solution and impact fields.
  - Added CSS components to _project-page.scss file to render value card with proper dimensions.
### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

before value card

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
after value

</details>

### CODE REVIEW

**Comment by t-will-gillis** [comment]
Review availability: Weds 12/7 3-5 pm
ETA: Weds 12/7 5pm

**Review by t-will-gillis — DISMISSED** [review]
Hi @patrickohh - 

Great job- It appears that the Project Value Card is rendering correctly- matching the proposed Figma screenshot-  for the five projects where `project.problem` exists and not the others with the Liquid if statememt. The new 'Value' card seems to be rendering and resizing correctly, and I don't see any unintended effects.  Good catch with the edits to the `_project-page.scss`-

**Review by MattPereira — CHANGES_REQUESTED** [review]
Hey @patrickohh solid work this looks really good! :+1: 

Just a couple of edits if possible:

- For small to medium screen sizes, I noticed the sustainable development goal card bumps up against the edge of its container card. Perhaps the CSS could be better adjusted for this screen size range ?<details><summary>See Screenshot</summary>
</details>

- For mobile screen sizes, the figma from the issue shows a modified design for the sustainable development goal card <details><summary>See Screenshot Comparison</summary><h3>Figma Design For Mobile</h3><h3>Current Pull Request on Mobile</h3>



</details>

**Comment by patrickohh** [comment]
Hi @MattPereira , I will look into making the necessary edits as soon as possible!

**Comment by patrickohh** [comment]
1. Progress: Haven't gotten a chance to look at the changes needed for this issue due to a busy week last week.
2. Blockers: Last week was just a filled schedule for me, but will take time this week to fix the needed issues.
3. Availability: 12/21 - 12/23
4. ETA: 12/23 Evening

**Comment by patrickohh** [comment]
> Hey @patrickohh solid work this looks really good! 👍
> 
> Just a couple of edits if possible:
> 
> * For small to medium screen sizes, I noticed the sustainable development goal card bumps up against the edge of its container card. Perhaps the CSS could be better adjusted for this screen size range ?See Screenshot
> 
> * For mobile screen sizes, the figma from the issue shows a modified design for the sustainable development goal card See Screenshot Comparison### Figma Design For Mobile### Current Pull Request on Mobile

Made the changes for the description card to avoid touching the outer div with small to medium screen sizes, however still finding trouble in implementing the mobile design for the description card. None of my media queries seem to work in the redesign. Is there a way to replace one div with another without having the change the whole structure of the html code?

File: _sass/components/_project-page.scss [hunk 1]
```diff
@@ -224,6 +224,95 @@
...
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  margin-right: auto;
+
+  // NEW MOBILE FRIENDLY RULES
+  @media #{$bp-below-mobile} {
+    width: 100%;
+  }
+}
```
Comment (MattPereira) [inline _sass/components/_project-page.scss]: I think you may want the media query here to be `$bp-below-tablet` for the `.sdg-description-card` to implement the the mobile design layout. I suggest adding `flex-direction:row` and `flex-wrap: nowrap` inside of a `below-tablet` media query to start moving in the right direction. The same basic pattern could probably be used to change the size and border radius of the `.sdg-img-item` to make it match the figma.

<details><summary>See Screenshot of mobile view if you make these changes</summary>

**Review by MattPereira — CHANGES_REQUESTED** [review]

File: _sass/components/_project-page.scss (hunk 1 above)
Comment (patrickohh) [reply _sass/components/_project-page.scss]: Ok will try to implement this later tonight. Thanks for the tip!

Comment (patrickohh) [reply _sass/components/_project-page.scss]: Made the necessary changes to implement the mobile design. Had to make a totally new div to create it more mobile-friendly.
<details><summary>See screenshot</summary>
value card mobile
</details>

**Review by jyaymie — DISMISSED** [review]
This is awesome work, @patrickohh! You developed a section that very closely matches the prototype and with logical class names and a good use of Liquid. I totally understand why you created a new div specifically for mobile as the shift in the elements' placement is tricky to implement here! That said, I think it is possible to work off just the one div you created. I played around with some of the code and got something working using this HTML structure:

```
<div class='sdg-description-card'>
     <h2 class='sdg-card-title title7'>Sustainable Development Goal</h2>
     <div class="sdg-description-wrapper">
          <img class='sdg-img' src='{{ page.sdg-image-src }}'/>
          <div class='sdg-description-grid-item'>{{ page.sdg }}</div>
     </div>
</div>
```

The main change is the wrapper around the SDG image and text. (If you have a better class name, by all means!) The SDG title, now on its own, should be more easier to manage when working on responsive design.

I'm gonna approve this PR because you did come up with a 100% valid solution, but I also know it's good practice to avoid repeated code. If you do attempt to refactor your code and get stuck, feel free to reach out on Slack!

**Review by MattPereira — DISMISSED** [review]
I concur with @jyaymie. The responsiveness from desktop down through mobile view is excellent. This is great work on a rather difficult Figma implementation 

Seems like all that remains is the trade off between shipping changes faster vs completely DRY code.

@patrickohh leaving the decision up to you. Let us know if you are done and we'll merge it as is or if you'd like to take on the challenge of implementing Jamie's clever optimization we are here for that too! :+1:

**Comment by patrickohh** [comment]
> This is awesome work, @patrickohh! You developed a section that very closely matches the prototype and with logical class names and a good use of Liquid. I totally understand why you created a new div specifically for mobile as the shift in the elements' placement is tricky to implement here! That said, I think it is possible to work off just the one div you created. I played around with some of the code and got something working using this HTML structure:
> 
> ```
> <div class='sdg-description-card'>
>      <h2 class='sdg-card-title title7'>Sustainable Development Goal</h2>
>      <div class="sdg-description-wrapper">
>           
>           <div class='sdg-description-grid-item'>{{ page.sdg }}</div>
>      </div>
> </div>
> ```
> 
> The main change is the wrapper around the SDG image and text. (If you have a better class name, by all means!) The SDG title, now on its own, should be more easier to manage when working on responsive design.
> 
> I'm gonna approve this PR because you did come up with a 100% valid solution, but I also know it's good practice to avoid repeated code. If you do attempt to refactor your code and get stuck, feel free to reach out on Slack!

Made some changes to use only one div, but running into an issue where the description section of the card wraps down too much from the screen size shrinking.

**Comment by patrickohh** [comment]
> I concur with @jyaymie. The responsiveness from desktop down through mobile view is excellent. This is great work on a rather difficult Figma implementation
> 
> Seems like all that remains is the trade off between shipping changes faster vs completely DRY code.
> 
> @patrickohh leaving the decision up to you. Let us know if you are done and we'll merge it as is or if you'd like to take on the challenge of implementing Jamie's clever optimization we are here for that too! 👍

Made an attempt on taking @jyaymie recommendation. But running into a format issue with the description section of the card.

File: _sass/components/_project-page.scss [hunk 2]
```diff
@@ -224,6 +224,107 @@
...
+.sdg-description-card {
+  padding-top: 30px;
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  margin-right: auto;
+  background: #02548B;
+  text-align: center;
+  border-radius: 10px 10px 11px 11px;
+  height: fit-content;
```
Comment (MattPereira) [inline _sass/components/_project-page.scss:272]: The Figma shows the description of the SDG card having a `text-align: start`.

I think the `text-align:center` applied to the `.sdg-description-card` is centering the text of the children elements.

Additionally, it looks like there is probably too much space on the bottom of the card compared to the Figma.

But I'm also not sure what you mean exactly when you say
> the description section of the card wraps down too much from the screen size shrinking

<details><summary>See Figma Design</summary>

</details>

<details><summary>See Current Pull Request</summary>

</details>

**Review by MattPereira — CHANGES_REQUESTED** [review]

File: _sass/components/_project-page.scss (hunk 2 above)
Comment (patrickohh) [reply _sass/components/_project-page.scss:272]: Sorry for the late reply. I have been pretty busy with other tasks and was gone all weekend. I will be sure to make these changes by the end of today!

Comment (patrickohh) [reply _sass/components/_project-page.scss:272]: @MattPereira Just implemented the change!

**Review by MattPereira — APPROVED** [review]
Superb work on this one @patrickohh :+1: 

We really appreciate your willingness to take on the challenge of refactoring the html and css to be more concise than the original implementation.

**Comment by patrickohh** [comment]
> Superb work on this one @patrickohh 👍
> 
> We really appreciate your willingness to take on the challenge of refactoring the html and css to be more concise than the original implementation.

Thank you for reviewing my code and offering suggestions along the way @jyaymie @MattPereira!

**Pull request merged**